### PR TITLE
bootstrapping with SSH

### PR DESCRIPTION
**Pull request by Loquacity**
https://github.com/SUSE/spacewalk/issues/11235

### CODE REVIEW

**Comment by Loquacity** [comment]
@jcayouette Note on the changelog entry: I figure if we add it in the PR, then it will merge when the PR merges, and we don't need to make a separate change (which should make it easier overall). I guess the real problem will be conflicts, but (at least in theory) they should be easy enough to manage. Let's see how it goes.

File: .changelog [hunk 1]
```diff
@@ -7,3 +7,4 @@
 
 * Added note about reserved characters in Proxy passwords(bsc#1134836)
 * Syntax and presentation updates to Salt Guide
+* Added Bootstrapping with SSH content
```
Comment (hustodemon) [inline .changelog:10]: I'd be specific, that this is about SSH key (bootstrapping via SSH (with password) was already supported before).
```suggestion
* Added Bootstrapping with SSH key content
```

File: modules/client-configuration/pages/salt-client-proxy.adoc [hunk 2]
```diff
@@ -19,7 +19,12 @@ For information about creating a bootstrap repository, see xref:client-configura
 By default, the SSH  port is [systemitem]``22``.
 . In the [guimenu]``User`` field, type the username to log in to the client.
 By default, the username is [systemitem]``root``.
-. In the [guimenu]``Password`` field, type password to log in to the client.
+. In the [guimenu]``Authentication Method`` field, select the authentication method to use for bootstrapping the client.
++
+* For password authentication, in the [guimenu]``Password`` field, type password to log in to the client.
+* For SSH Private key authentication, enter the private key and the associated passphrase.
+The key is only stored for as long as the bootstrapping process takes to complete.
+The key is deleted after bootstrapping is successful.
```
Comment (hustodemon) [inline modules/client-configuration/pages/salt-client-proxy.adoc:27]: Nitpick: the key is also deleted, when the bootstrapping is unsuccessful :-), so either I'd suggest using something more general ("..the key is deleted after bootstrapping"), or I'd even leave this sentence out (the line 26 already says that).

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 3]
```diff
@@ -20,8 +20,20 @@ User::
...
+// I've added these new bits in the existing format for now, but this whole section will get overhauled with the Ref Guide updates. LKB --2020-04-17
+Authentication Method::
+Select either ``Password`` or ``SSH Private Key``.
+
 Password::
-Input the clients login password.
+For password authentication, enter the client's login password.
+
+SSH Private Key::
+For SSH authentication, copy the SSH private key.
```
Comment (hustodemon) [inline modules/reference/pages/systems/bootstrapping.adoc:31]:
```suggestion
For SSH key authentication, copy the SSH private key.
```
As mentioned in the changelog, I'd be explicit about the key here (the "SSH authentication" happens also in the password case).

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 4]
```diff
@@ -20,8 +20,20 @@ User::
...
+The key is only stored for as long as the bootstrapping process takes to complete.
+The key is deleted after bootstrapping is successful.
```
Comment (hustodemon) [inline modules/reference/pages/systems/bootstrapping.adoc:33]: Same as above ^^^.

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 5]
```diff
@@ -20,8 +20,20 @@ User::
...
+
+SSH Private Key Passphrase::
+For SSH authentication, copy the passphrase for the private key.
```
Comment (hustodemon) [inline modules/reference/pages/systems/bootstrapping.adoc:36]: I'd say "enter" - the passphrase is basically a password, which users type in.
```suggestion
For SSH authentication, enter the passphrase for the private key.
```

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 6]
```diff
@@ -30,6 +42,7 @@ Disable SSH Strict Key Host Checking::
 This check box is selected by default.
 This allows the script to auto-accept host keys without requiring a user to manually authenticate.
 
+// Aside from the terrible formatting here, does this need the tech preview notice removed? --LKB 2020-04-17
```
Comment (hustodemon) [inline modules/reference/pages/systems/bootstrapping.adoc:45]: Wow, I didn't know that we still have "minion-less" salt clients as a tech preview. Is this true @renner ?

Comment (Loquacity) [reply modules/reference/pages/systems/bootstrapping.adoc:45]: I suspect it's more that we failed to take the note out some time ago, honestly ...

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 7]
```diff
@@ -30,11 +41,9 @@ Disable SSH Strict Key Host Checking::
 This check box is selected by default.
 This allows the script to auto-accept host keys without requiring a user to manually authenticate.
 
+
```
Comment (keichwa) [inline modules/reference/pages/systems/bootstrapping.adoc:44]:
```suggestion
```

File: modules/reference/pages/systems/bootstrapping.adoc [hunk 8]
```diff
@@ -30,11 +41,9 @@ Disable SSH Strict Key Host Checking::
...
 Manage System Completely via SSH (Will not Install an Agent)::
 +
```
Comment (keichwa) [inline modules/reference/pages/systems/bootstrapping.adoc:46]:
```suggestion
```
I think + i not needed here...

**Pull request merged**